manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/builder/DefaultPortsBuilder.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.generator.services.builder;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.Sets;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.utils.NormalizationUtil;
+import io.hyscale.generator.services.constants.ManifestGenConstants;
+import io.hyscale.generator.services.exception.ManifestErrorCodes;
+import io.hyscale.generator.services.model.ManifestResource;
+import io.hyscale.plugin.framework.models.ManifestSnippet;
+import io.hyscale.plugin.framework.util.GsonSnippetConvertor;
+import io.hyscale.servicespec.commons.model.service.Port;
+import io.kubernetes.client.custom.IntOrString;
+import io.kubernetes.client.openapi.models.V1ContainerPort;
+import io.kubernetes.client.openapi.models.V1ServicePort;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This class is to generate the default ports taken from the
+ * service spec for the manifest snippet.
+ * Handles Ports for Services as well as Agents.
+ */
+@Component
+public class DefaultPortsBuilder {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultPortsBuilder.class);
+
+    public List<ManifestSnippet> generatePortsManifest(List<Port> portList, String podSpecOwner) throws HyscaleException {
+        return generatePortsManifest(portList, podSpecOwner, 0);
+    }
+
+    public List<ManifestSnippet> generatePortsManifest(List<Port> portList, String podSpecOwner, int containerIndex) throws HyscaleException {
+        List<ManifestSnippet> manifestSnippetList = new ArrayList<>();
+        if (portList != null && !portList.isEmpty()) {
```

Comment:
Use CollectionUtils.isEmpty and return if true to avoid nesting of code inside if